core/src/worker/mod.rs

```diff
@@ -340,15 +343,16 @@ impl Worker {
     /// completed
     async fn shutdown(&self) {
```

Comment:
We should just get rid of this `shutdown` method.
It's basically finalize at this point.

Reply:
Yeah, I wanted to do that anyway. If we're fine with that then I'll get rid of it.

Reply:
Though, that said, it's quite annoying to change for many of the tests. Many of them like to see when shutdown completes while doing something else, and that can't be done with finalize since it consumes.

It's probably worth me reconciling all that, though... but I'm not sure we want to do it right now as it's a decent chunk of work.